.github/workflows/astarte-apps-build-workflow.yaml

```diff
@@ -0,0 +1,155 @@
+name: Build and Test Astarte Apps
+
+on:
+  # Run when pushing to stable branches
+  push:
+    paths:
+    - 'apps/**'
+    - '.github/workflows/astarte-apps-build-workflow.yaml'
+    branches:
+    - 'master'
+    - 'release-*'
+  # Run on branch/tag creation
+  create:
+  # Run on pull requests matching apps
+  pull_request:
+    paths:
+    - 'apps/**'
+    - '.github/**'
+
+env:
+  elixir_version: 1.8.1
+  otp_version: 21.3
+
+jobs:
+  test-dialyzer:
+    name: Check Dialyzer
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        app: [astarte_appengine_api, astarte_data_updater_plant, astarte_trigger_engine, astarte_pairing, astarte_pairing_api, astarte_realm_management, astarte_realm_management_api, astarte_housekeeping, astarte_housekeeping_api]
```

Comment:
Shouldn't this be
```
- astarte_appengine_api
- astarte_data_updater_plant
...
```
since we're using yaml? This would also make it more readable